bin/metrics-zookeeper-cluster.rb

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env ruby
+#
+# metrics-zookeeper.rb
+#
+# Collect ZooKeeper metrics
+# ===
+#
+# DESCRIPTION:
+# This plugin gathers metrics from an Exhibitor run ZooKeeper cluster,
+# based on the collectd plugin:
+#
+#   https://github.com/Nextdoor/collectd_plugins/blob/master/zookeeper/zookeeper.sh
+#
+#
+# PLATFORMS:
+#   Linux, BSD, Solaris
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# LICENSE:
+# Sean Clemmer sczizzo@gmail.com
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+#  for details.
+#
+
+require 'sensu-plugin/metric/cli'
+require 'socket'
+require 'net/http'
+require 'json'
+
+class ZookeeperMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  option :exhibitor,
+         description: 'exhibitor end node for status checks',
+         short: '-e Exhibitor status end point',
+         long: '--exhibitor status end point',
+         default: 'http://localhost/exhibitor/v1/cluster/status'
+
+  option :scheme,
+         description: 'Metric naming scheme, text to prepend to metrics',
+         long: '--scheme SCHEME',
+         default: 'zookeeper'
+
+  def follow_url(uri_str, agent = 'curl/7.43.0', max_attempts = 10, timeout = 10)
```

Comment:
why is the default User Agent curl?

Reply:
Actually there is no reason to choose or not to choose curl. The gist I used to have a proper url_follow has this one. https://gist.github.com/sekrett/7dd4177d6c87cf8265cd

But if we actually consider this, having a url follow also does not make sense. This end point should be exact, and exhibitor normally does not url redirect. A reason to have one could be a reverse proxy, which would imply there is a network separation in place, and this may also mean, the zookeeper instances may not be reachable.

Reply:
I have not touched zookeeper in a few years, originally the redirect I had in mind was indeed a reverse proxy. Is that really uncommon in the zookeeper space? Regarding default user agent maybe something like `sensu-check` makes sense. I run many things behind an nginx reverse proxy and I do block requests from user agents that are empty and other scenarios so having a reasonable UA makes sense to me.

Reply:
One last note regarding an nginx proxy: another reason for redirection might be to redirect from http -> https.

Reply:
I don't feel super strongly about this but given the effort of changing curl to `sensu-check` seems worth it to give you better access logs.

Reply:
I have not heard from you, I will merge and update this on release.